source/memory/memory.go

```diff
@@ -113,6 +129,34 @@ func (s *memorySource) GetFeatures() *nfdv1alpha1.Features {
 	return s.features
 }
 
+// detectSwap detects Swap node information
+func detectSwap() (map[string]string, error) {
+	procBasePath := hostpath.ProcDir.Path("swaps")
+	file, err := os.Open(procBasePath)
+	defer func() {
+		err = file.Close()
+		if err != nil {
+			klog.ErrorS(err, "unable to close swaps file")
+		}
+	}()
+	if err != nil {
+		return nil, err
+	}
+
+	// Read the CSV data
+	reader := csv.NewReader(file)
+	reader.FieldsPerRecord = -1 // Allow variable number of fields
+	data, err := reader.ReadAll()
+	if err != nil {
+		return nil, err
+	}
+	// /proc/swaps has a header row
+	// If there is more than a header then we assume we have swap.
+	return map[string]string{
+		"is_swap": strconv.FormatBool(len(data[0]) > 1),
```

Comment:
How about "enabled"? Then the feature to look for would be `memory.swap.enable` which sounds more logical to me than `memory.swap.is_swap`. WDYT?

Reply:
good idea. updated.